tensorflow/core/kernels/fused_batch_norm_op.cc

```diff
@@ -245,7 +369,8 @@ struct FusedBatchNorm<GPUDevice, T, U> {
                   const Tensor& estimated_variance, U epsilon, Tensor* y,
                   Tensor* batch_mean, Tensor* batch_var, Tensor* saved_mean,
                   Tensor* saved_inv_var, TensorFormat tensor_format,
-                  bool is_training) {
+                  void* reserve_space_allocator,
```

Comment:
Please use ScratchAllocator* as the type, for both gpu and cpu implementation.

Reply:
The problem is the ScratchAllocator is defined in stream_executor.h. For the some non-GPU tests (Something about **Adroid**, sorry I forget the name and it hides now, but I think you can see the test name), it shows the stream_executor.h cannot be found. So, my understanding is for the non-GPU environment, we cannot rely on stream_executor.h. Please correct me if I am wrong.

Reply:
Oh, I think I found the log. For the` //tensorflow/tools/ci_build/builds:gen_android_out`, it prompts `tensorflow/core/kernels/fused_batch_norm_op.cc:32:10: fatal error: 'tensorflow/core/platform/stream_executor.h' file not found`.

Reply:
@aaroey Any suggestion about this?

Reply:
I'll try locally and get back to you soon.

Reply:
The reason is android builds use a different target that depends on fused_batch_norm_op.cc, not `:fused_batch_norm_op` in `kernels/BUILD`. In that case, I think it may not be a good idea to add stream_executor dependency to android builds. Instead please add a dummy class `ScratchAllocator` when `#if !GOOGLE_CUDA` to fused_batch_norm_op.cc, add a comment about why adding it, and make the dummy CudnnBatchNormAllocatorInTemp and CudnnBatchNormAllocatorInOutput inherit from it. Also maybe revert (most?) of the changes of the last two commits so we have the explicit type and other benefits back.

Another thing is please use `if_cuda()` to guard the newly added stream executor dependency of target `fused_batch_norm_op`.